astropy/io/fits/src/compressionmodule.c

```diff
@@ -888,7 +899,7 @@ PyObject* compression_compress_hdu(PyObject* self, PyObject* args)
     PyObject* retval = NULL;
     tcolumn* columns = NULL;
 
-    void* outbuf;
+    void* outbuf = NULL;
```

Comment:
Yes, it turned out that was in fact the most important change I still had to make -- the problem is that `init_output_buffer` does not touch `outbuf` at all when an error occurs and then the comparison with `NULL` fails, leading to a later trial to unmap `buf`.

(I guess an alternative would be to check for `PyErr_Occurred` or to let `init_output_buffer` return or set a status flag.)